packages/opentelemetry-instrumentation-llamaindex/tests/test_agents.py

```diff
@@ -23,17 +24,49 @@ def multiply(a: int, b: int) -> int:
     spans = exporter.get_finished_spans()
 
     assert {
-        "ReActAgent.agent",
-        "FunctionTool.tool",
-        "openai.chat",
-    } == set([span.name for span in spans])
-
-    agent_span = next(span for span in spans if span.name == "ReActAgent.agent")
-    function_tool = next(span for span in spans if span.name == "FunctionTool.tool")
-    openai_span = next(span for span in spans if span.name == "openai.chat")
+        "agent_step.llama_index.workflow",
+        "llm.llama_index.task",
+        "function_call.llama_index.task",
```

Comment:
looks like openai is missing here?

Reply:
I am not sure if that is the case: the llm calls are under llm.llama_index.task, e.g. see L40: `assert llm_span_1.attributes[SpanAttributes.LLM_REQUEST_MODEL] == "gpt-3.5-turbo-0613"`.

Imo, there is the question if the OpenAi instrumentor is needed if one uses langchain. On one side not because langchain is supposed to take care of that. On the other hand, if one explicitly uses the OpenAI instrumentor, then the expectation is to see these traces.

Reply:
After some digging: openai/shared/chat_wrappers gets called, but context_api.get_value(SUPPRESS_LANGUAGE_MODEL_INSTRUMENTATION_KEY) is True - this was added recently, wasn't it? I wonder if it's just a new env variable which needs to be set in conftest...

Reply:
This is it - and I copied it: https://github.com/traceloop/openllmetry/pull/1452/files#diff-b5474fc40399d3e9320e536a98b121519743b68e5b7271884a8b853052a5ffe8R273

@nirga Do you remember why you are setting this value?

Reply:
I replaced it now with current_context. Now the LLM spans look ok. There is even the test_chroma_vector_store where there are 2 spans outside of the engine - which is correct imho.

Reply:
The reason I did that is because we report an OpenAI span now within LangChain so I didn't want to report the same data twice (if using LangChain). It's only turned on within LangChain but otherwise defaults to off. Not sure I understood what you've done differently to make it work?

Reply:
This is only about your comment "looks like openai is missing here?". My understanding is that langchain and llama are both frameworks, but apparently they have some differences, e.g. how they handle openai calls.

Regarding the bigger issue in #1541: that's still not solved.